### PR TITLE
Further improvements to config check devicegw

### PR DESCRIPTION
**Pull request by DunderRoffe**

### CODE REVIEW

File: libsoftwarecontainer/src/gateway/devicenodegateway.cpp [hunk 1]
```diff
@@ -38,18 +38,31 @@ ReturnCode DeviceNodeGateway::readConfigElement(const json_t *element)
...
+
+        if (!minorSpecified) {
+            log_error() << "Minor version not specified but major version is. " << notAllowed;
+            return ReturnCode::Failure;
+        }
+
+        if (!modeSpecified) {
+            log_error() << "Mode has to be specified when minor and major version is.";
+            return ReturnCode::Failure;
+        }
```
Comment (frznlogic) [inline libsoftwarecontainer/src/gateway/devicenodegateway.cpp:53]: This is very chatty code, might be rewritten to simplify it? Possibly just a 

const bool xxxSpecified = checkSpecified(dev.xxx, "xxx");

?

File: libsoftwarecontainer/src/gateway/devicenodegateway.cpp [hunk 2]
```diff
@@ -62,11 +75,11 @@ bool DeviceNodeGateway::activateGateway()
 {
     for (auto &dev : m_devList) {
         log_info() << "Mapping device " << dev.name;
-        const int mode = std::atoi(dev.mode.c_str());
 
         if (dev.major.length() != 0) {
             const int majorVersion = std::atoi(dev.major.c_str());
             const int minorVersion = std::atoi(dev.minor.c_str());
+            const int mode = std::atoi(dev.mode.c_str());
```
Comment (tobsan) [inline libsoftwarecontainer/src/gateway/devicenodegateway.cpp]: Do we really want to use atoi or something more failsafe?

Comment (frznlogic) [reply libsoftwarecontainer/src/gateway/devicenodegateway.cpp]: std::stoi might be better i guess? Is that enough?

Comment (DunderRoffe) [reply libsoftwarecontainer/src/gateway/devicenodegateway.cpp]: Solved this by using parseInt that is already used in the agent so moved it to comon.

**Pull request merged**